contrib_bots/lib/CommuteBot/CommuteBot.py

```diff
@@ -0,0 +1,219 @@
+        To add more than 1 input for a category,
+        e.g. more than 1 destinations,
+        use (|), e.g. destinations=Trump+Tower|The+White+House
+        No spaces within addresses.
+        Departure times and arrival times must be in the UTC timezone,
+        you can use the timezone bot.
+                            '''
+            self.send_info(message, help_info, client)
+        else:
+            params = {}
+            # converts input parameters for api into a dictionary
+            # structure for more aethestic coding
+            for item in content_list:
+                # enables key value pair
+                org = item.split('=')
+
+                # ensures that invalid inputs are not entered into url request
+                if len(org) != 2:
+                    continue
+
+                key = org[0]
+                value = org[1]
+                params.update({key: value})
+
+            # limited to UTC timezone because of difficulty with user inputting
+            # correct string for input
+            if 'departure_time' in params:
+                if params['departure_time'] != 'now':
+                    departure_time = params['departure_time']
+                    times = departure_time.split(',')
+                    times = [int(str(x)) for x in times]
+                    a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                    b = dt.datetime(times[0],
+                                    times[1],
+                                    times[2],
+                                    times[3],
+                                    times[4],
+                                    times[5])
+                    total_seconds = str(int((b-a).total_seconds()))
+                    params.update({'departure_time': total_seconds})
+
+            if 'arrival_time' in params:
+                arrival = params['arrival_time']
+                times = arrival.split(',')
+                times = [int(str(x)) for x in times]
+                a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                b = dt.datetime(times[0],
+                                times[1],
+                                times[2],
+                                times[3],
+                                times[4],
+                                times[5])
+                total_seconds = str(int((b-a).total_seconds()))
+                params.update({'arrival_time': total_seconds})
+
+            # adds API Authentication Key to url request
+            config = cp.ConfigParser()
+            settings = open('lib/settings.ini')
+            config.readfp(settings)
+            apik = config.get('Google.com', 'apik')
+            params.update({'key': apik})
+
+            # sends url api request and converts response into json format
+            r = (requests.get('https://maps.googleapis.com/maps/api/'
+                              + 'distancematrix/json',
+                              params=params))
+            rjson = r.json()
```

Comment:
There is possibility that the response is not OK, because of user or server errors. You should check for those before attempting to parse a JSON.